node/src/WorkQueue.ts

```diff
@@ -47,6 +47,35 @@ export class WorkQueue {
     pipeline.lpush(this.mainQueueKey, item.id);
   }
 
+  /**
+   * Add an item to the work queue in a atomic way.
+   * This creates a pipeline and executes it on the database if the item id is not already in either the main queue or the procesing queue.
+   *
+   * @param {Redis} db The Redis Connection.
+   * @param item The item that will be executed using the method addItemToPipeline.
+   */
+  async addAtomicItem(db: Redis, item: Item): Promise<boolean | null> {
+    return new Promise<boolean | null>((resolve, reject) => {
+      db.multi()
+        .lrange(this.mainQueueKey, 0, -1)
+        .lrange(this.processingKey, 0, -1)
+        .exec(async (err, results) => {
+          if (
+            results &&
+            ((results[0][1] as string[]).includes(item.id) ||
+              (results[1][1] as string[]).includes(item.id))
+          ) {
+            resolve(null);
+          } else {
+            const pipeline = db.pipeline();
+            this.addItemToPipeline(pipeline, item);
+            await pipeline.exec();
+            resolve(true);
+          }
+        });
+    });
```

Comment:
Can we use `await` here on the first transaction to neaten things up?

Also see the comment about insertion between the transactions from the Go code.